File: day-1/day-1-part-2.py
counter = 0

with open('../src/main/scala/Day1/input.txt') as f:
    lines = [int(line.rstrip()) for line in f.readlines()]
    curr_sum = lines[:3]
    for line in lines[3:]:
        prev_sum = sum(curr_sum)

        # new sum
        curr_sum.pop(0)
        curr_sum.append(line)

        if sum(curr_sum) > prev_sum:
            counter += 1

print(counter)

File: advent-of-code-2021/src/main/scala/Day1/Day1Part2.scala
package Day1

object Day1Part2 extends App {
    val source = io.Source.fromFile("./src/main/scala/Day1/input.txt")
    val lines = try source.getLines().toList finally source.close()

    val num_inc = lines
        .map(line => line.toInt)
        .sliding(4)
        .map(l => if (l.takeRight(3).sum > l.take(3).sum) 1 else 0)
        .sum

    print(num_inc)
}
